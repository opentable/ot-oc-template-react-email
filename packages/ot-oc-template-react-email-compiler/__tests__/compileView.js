jasmine.DEFAULT_TIMEOUT_INTERVAL = 50000;

const path = require("path");
const fs = require("fs-extra");
const compileView = require("../lib/compileView.js");

test("valid component", done => {
  const componentPath = path.join(__dirname, "../../../mocks/react-component");
  const publishPath = path.join(componentPath, "_packageCompileViewTest");
  const publishFileName = "template.js";

  const options = {
    componentPackage: fs.readJsonSync(`${componentPath}/package.json`),
    componentPath,
    publishPath,
    publishFileName,
    production: true
  };

  compileView(options, (err, compiledViewInfo) => {
    expect(compiledViewInfo).toMatchSnapshot();
    expect(
      fs.readFileSync(path.join(publishPath, publishFileName), "UTF8")
    ).toMatchSnapshot();
    fs.removeSync(publishPath);
    done();
  });
});

test("invalid component", done => {
  const componentPath = path.join(
    __dirname,
    "../../../mocks/invalid-jsx-component"
  );
  const publishPath = path.join(componentPath, "_packageCompileViewTest2");
  const publishFileName = "template.js";

  const options = {
    componentPackage: fs.readJsonSync(`${componentPath}/package.json`),
    componentPath,
    publishPath,
    publishFileName,
    production: true
  };

  compileView(options, (err, compiledViewInfo) => {
    expect(err).toContain(
      "Adjacent JSX elements must be wrapped in an enclosing tag"
    );
    done();
  });
});
