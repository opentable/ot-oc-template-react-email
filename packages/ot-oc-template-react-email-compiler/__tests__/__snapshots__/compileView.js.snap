// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`valid component 1`] = `
Object {
  "bundle": Object {
    "hashKey": "95006dbf0f72b31905bce0cc0fc87067877f059f",
  },
  "template": Object {
    "hashKey": "645ac3df91dad323fe217aaf1f61fb70f63d23f8",
    "src": "template.js",
    "type": "ot-oc-template-react-email",
  },
}
`;

exports[`valid component 2`] = `"var oc=oc||{};oc.components=oc.components||{};oc.components['645ac3df91dad323fe217aaf1f61fb70f63d23f8']=function(model){ return model.__html ? model.__html : ''; }"`;
